src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -510,6 +511,42 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
         n->recalc_output_layout(true);
     }
 
+    const auto reorder_input_pooling = [&p, &rf](typed_program_node<pooling>& pooling_node) {
+        // Change input data type of pooling node from i32 to f32
+        auto& input = pooling_node.input();
+        auto input_layout = input.get_output_layout();
+        if (input_layout.data_type == data_types::i32) {
+            auto new_layout = input_layout;
+            new_layout.data_type = data_types::f32;
+            auto new_input = rf.get_reorder(input.id(), input_layout, new_layout);
+            if (new_input.first) {
+               p.add_intermediate(new_input.first, pooling_node, 0, !new_input.second);
+               p.get_or_create(new_input.first).recalc_output_layout(true);
+            }
+        }
+
+        // Change weights type i32 to f32
+        if (pooling_node.get_dependencies().size() >= 2) {
+            auto& weights = pooling_node.argmax();
+            auto weights_layout = weights.get_output_layout();
+            if (weights_layout.data_type == data_types::i32) {
+                auto new_layout = weights_layout;
+                new_layout.data_type = data_types::f32;
+                auto new_input = rf.get_reorder(weights.id(), weights_layout, new_layout);
+                if (new_input.first) {
+                    p.add_intermediate(new_input.first, pooling_node, 1, !new_input.second);
+                    p.get_or_create(new_input.first).recalc_output_layout(true);
+                }
+            }
+        }
+    };
+
+    for (auto& prim : p.get_processing_order()) {
+        program_helpers::do_for_types<pooling>(
+            *prim,
+            reorder_input_pooling);
+    }
```

Comment:
I think you can do this in here https://github.com/openvinotoolkit/openvino/blob/52434b7ea0e18dddff72ccb802b5f5c3e3bdafa0/src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp#L728

Reply:
done